integration/src/integration/java/pl/allegro/tech/hermes/integration/MessagesPublishingStartupValidationTest.java

```diff
@@ -0,0 +1,131 @@
+package pl.allegro.tech.hermes.integration;
+
+import com.google.common.io.Files;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.common.config.Configs;
+import pl.allegro.tech.hermes.frontend.server.PublishingStartupValidationException;
+import pl.allegro.tech.hermes.integration.env.CustomKafkaStarter;
+import pl.allegro.tech.hermes.integration.env.FrontendStarter;
+import pl.allegro.tech.hermes.integration.env.ManagementStarter;
+import pl.allegro.tech.hermes.test.helper.builder.TopicBuilder;
+import pl.allegro.tech.hermes.test.helper.endpoint.HermesAPIOperations;
+import pl.allegro.tech.hermes.test.helper.endpoint.HermesEndpoints;
+import pl.allegro.tech.hermes.test.helper.endpoint.JerseyClientFactory;
+import pl.allegro.tech.hermes.test.helper.environment.KafkaStarter;
+import pl.allegro.tech.hermes.test.helper.environment.ZookeeperStarter;
+import pl.allegro.tech.hermes.test.helper.util.Ports;
+
+import javax.ws.rs.client.WebTarget;
+
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static pl.allegro.tech.hermes.integration.test.HermesAssertions.assertThat;
+
+public class MessagesPublishingStartupValidationTest extends IntegrationTest {
+
+    private static final String BROKER_MESSAGES_PUBLISHING_VALIDATION_GROUP = "someRandomGroup";
+    private static final String BROKER_MESSAGES_PUBLISHING_VALIDATION_TOPIC = "someRandomTopic";
+    private static final int ZOOKEEPER_PORT = 14194;
+    private static final String ZOOKEEPER_URL = "localhost:" + ZOOKEEPER_PORT;
+    private static final int MANAGEMENT_PORT = 18084;
+    private static final String MANAGEMENT_URL = "http://localhost:" + MANAGEMENT_PORT + "/";
+    private static final int KAFKA_PORT = 9098;
+    private static final String KAFKA_URL = "localhost:" + KAFKA_PORT;
+
+    private ZookeeperStarter zookeeper;
+    private HermesAPIOperations operations;
+    private ManagementStarter hermesManagement;
+    private KafkaStarter kafkaStarter;
+
+    @BeforeClass
+    public void setupEnvironment() throws Exception {
+        zookeeper = setupZookeeper();
+        kafkaStarter = setupKafka();
+        hermesManagement = setupHermesManagement();
+        operations = setupOperations();
+    }
+
+    @AfterClass
+    public void cleanEnvironment() throws Exception {
+        hermesManagement.stop();
+        kafkaStarter.stop();
+        zookeeper.stop();
+    }
+
+    @Test
+    public void shouldReturnCorrectReadinessStatusBasedOnKafkaReadiness() throws Exception {
```

Comment:
This test should have name indicating it is connected to status health of hermes frontend. There is no readiness concept yet :)